### ISSUE TITLE
Use the # sign to locate the URL of some properties that are not easy to reconcile with Wikidata(to be clarified)

### ISSUE THREAD

**Issue by candlecao** [DDMAL/linkedmusic-datalake#107]
For example,
https://musicbrainz.org/doc/Event#Cancelled.

**Comment by ahankinson** [comment]
Why did you choose the document fragment delimiter?

**Comment by dchiller** [comment]
I understand this is only an example, but I'm also not sure why, if you wanted to reconcile an event status as being cancelled, this would be hard to do in Wikidata?

**Comment by candlecao** [comment]
> Why did you choose the document fragment delimiter?

@ahankinson 
Afaik, (1)The method is common in the area of linked data and information management. (2) In practice of Semantic Web, MusicBrainz itself is just a well-recognized metadata metadata schema. Many ontologies reuse it directly(see the picture).(3)The # symbol can directly locate to the attribute "Cancelled" on the page https://musicbrainz.org/doc/Event.
![Screenshot 2024-07-29 at 11 10 07](https://github.com/user-attachments/assets/d8078297-702e-4585-835d-f568a800ec8b)


@dchiller 
It's not easy to find an exact match. Therefore, we can directly use the https://musicbrainz.org/doc/Event#Cancelled.

I am not necessarily surely right. It's just based on my experience.
Thank you for suggestions.

**Comment by dchiller** [comment]
> It's not easy to find an exact match.

https://www.wikidata.org/wiki/Q114342413 ?

Or if there isn't one we create it? I feel like we should be very, very intentional and careful about departing from the "Wikidata is our schema" principle. And it seems fishy to me that we would want to represent an object as a URI fragment.

**Comment by ahankinson** [comment]
The document fragment component is a complicated identifier form to use correctly. 

HTTP clients never send the fragment portion of a URL to the server. Thus if you need to look something up on the server based on the fragment identifier you can’t. 

The fragment will prompt browsers to jump to that portion of the page if it is returned as HTML, but this changes if the output form of the HTML changes. Thus it isn’t really that great as a persistent identifier.

If you are only using it for a URI and never expect it to be retrievable, then it is fine as a CURIE value. So you should make sure your only purpose for using this is to have a unique URI.

**Comment by candlecao** [comment]
> > It's not easy to find an exact match.
> 
> https://www.wikidata.org/wiki/Q114342413 ?
> 
> Or if there isn't one we create it? I feel like we should be very, very intentional and careful about departing from the "Wikidata is our schema" principle. And it seems fishy to me that we would want to represent an object as a URI fragment.

I believe "https://www.wikidata.org/wiki/Q114342413" is definitely not feasible. It's an entity (an instance or a class), rather than a property. A property’s URI typically contains “P” (e.g., …P…) rather than “Q” (e.g., …Q…).

I recall that ChatGPT understands the distinction between P and Q. So we had better adhere to this differentiation or at least not confuse the two. Subsequently, maintaining this distinction will significantly aid ChatGPT in converting natural language questions into SPARQL queries through unified reconciliation.


departing from the "Wikidata is our schema" principle--it's another topic we may talk someday.

**Comment by candlecao** [comment]
> The document fragment component is a complicated identifier form to use correctly.
> 
> HTTP clients never send the fragment portion of a URL to the server. Thus if you need to look something up on the server based on the fragment identifier you can’t.
> 
> The fragment will prompt browsers to jump to that portion of the page if it is returned as HTML, but this changes if the output form of the HTML changes. Thus it isn’t really that great as a persistent identifier.
> 
> If you are only using it for a URI and never expect it to be retrievable, then it is fine as a CURIE value. So you should make sure your only purpose for using this is to have a unique URI.

OK, thank you. I got it a lot . At present, our purpose for using this is to have a unique URI. @Yueqiao12Zhang, for your reference too.

**Comment by fujinaga** [comment]
What about this?
URI_of_an_event(e.g., concert) wdt:P793 wd:Q114342413

**Comment by dchiller** [comment]
> I believe "https://www.wikidata.org/wiki/Q114342413" is definitely not feasible. It's an entity (an instance or a class), rather than a property. A property’s URI typically contains “P” (e.g., …P…) rather than “Q” (e.g., …Q…).

Right, but what's the "property" you are trying to represent here? I mean, here's a way I could represent a cancelled event:

```mermaid
graph TD
   A[https://musicbrainz.org/event/d801c6c8-7870-4506-9709-af3890bf1e74] --> B[P31: is instance of]
   B --> C[Q114342413]
```

I'm not sure what the property would be of what you are doing? This? Why?

```mermaid
graph TD
   A[https://musicbrainz.org/event/d801c6c8-7870-4506-9709-af3890bf1e74] --> B["has cancelled event status"]
   B --> C["True"]
```

**Comment by Yueqiao12Zhang** [comment]
> > I believe "https://www.wikidata.org/wiki/Q114342413" is definitely not feasible. It's an entity (an instance or a class), rather than a property. A property’s URI typically contains “P” (e.g., …P…) rather than “Q” (e.g., …Q…).
> 
> Right, but what's the "property" you are trying to represent here? I mean, here's a way I could represent a cancelled event:
> 
> ```mermaid
> graph TD
>    A[https://musicbrainz.org/event/d801c6c8-7870-4506-9709-af3890bf1e74] --> B[P31: is instance of]
>    B --> C[Q114342413]
> ```

I agree with this one. It makes sense. However, since in the original JSON, the object "cancelled" can only have a boolean value. We can only modify the conversion script code to customize it. Adding some specific customization to the conversion might cause bugs.

**Comment by fujinaga** [comment]
The whole point of this exercise is to find out what needs conversion customization and what can be automatically converted with OpenRefine. What's important is to identify the challenges, such as this example.

**Comment by ahankinson** [comment]
Will you need to model cancelled events that are not in musicbrainz?

**Comment by fujinaga** [comment]
> Will you need to model cancelled events that are not in musicbrainz?

Yes, for example, thesession.org also has a cancelled field.

**Comment by ahankinson** [comment]
So probably a more generic but pre-defined ontology like Wikidata is more suitable than a bespoke vocabulary from MusicBrainz?

**Comment by fujinaga** [comment]
Agreed. So let's go with this one:
```mermaid
graph TD
   A[https://musicbrainz.org/event/d801c6c8-7870-4506-9709-af3890bf1e74] --> B[P31: is instance of]
   B --> C[Q114342413]

**Comment by candlecao** [comment]
OK, to confirm: for all entities categorized as events that have an attribute indicating whether they are "cancelled or not cancelled", we can temporarily apply the following pattern:
`<an event> wdt:P31 <wd:Q114342413> .`
We may draft a principle of all the reconciliation, especially for particular case like this. @fujinaga 
@Yueqiao12Zhang

**Comment by candlecao** [comment]
> So probably a more generic but pre-defined ontology like Wikidata is more suitable than a bespoke vocabulary from MusicBrainz?

Yes, I basically agree.

**Comment by SCN-MNG** [comment]
Issue completed.

The current consensus that we will create new Wikidata properties/entities if needed.
